build/android/native/build.gradle

```diff
@@ -0,0 +1,60 @@
+apply plugin: 'com.android.library'
+import org.ajoberstar.grgit.Grgit
+
+android {
+	compileSdkVersion 29
+	buildToolsVersion '29.0.2'
+	defaultConfig {
+		minSdkVersion 16
+		targetSdkVersion 29
+		externalNativeBuild {
+			ndkBuild {
+				arguments '-j10',
+						'APP_STL=c++_shared',
+						'COMPILER_VERSION=clang',
+						'APP_SHORT_COMMANDS=true',
+						"versionMajor=${versionMajor}",
+						"versionMinor=${versionMinor}",
+						"versionPatch=${versionPatch}"
+			}
+		}
+	}
+
+	externalNativeBuild {
+		ndkBuild {
+			path file('jni/Android.mk')
+		}
+	}
+
+	// supported architectures
+	splits {
+		abi {
+			enable true
+			reset()
+			include 'armeabi-v7a', 'arm64-v8a'//, 'x86'
+		}
+	}
+
+	buildTypes {
+		release {
+			externalNativeBuild {
+				ndkBuild {
+					arguments 'NDEBUG=1'
+				}
+			}
+		}
+	}
+}
+
+task cloneGitRepo() {
+	def destination = file('deps')
+	if(!destination.exists()) {
+		def grgit = Grgit.clone(
+				dir: destination,
+				uri: 'https://github.com/MultiCraft/deps'
```

Comment:
This will need to be changed to point to a repo in the minetest organisation before merging

Reply:
If you want to take care of this - always welcome. Button for fork in the upper right

Reply:
Why exactly is this needed? Don't normal installed libraries suffice for compiling?

Reply:
Sorry, but where do you think Windows will get "normal installed libraries" if building them on it does not seem possible?

Reply:
For my old Win32 Minetest builds I downloaded the libraries manually. Then I either compiled them or downloaded the pre-compiled binaries. The linkage to Minetest happened in CMake.
This PR does not make use of CMake. Would it be possible to add the library requirements to a specific gradle file?

Reply:
Android uses Gradle. Google recommends using Android.mk. Although Gradle can work with CMake.
Switching to CMake assembly is possible, although I’m not sure that haters of Android assembly want to see so many Android-specific things in CMakeLists.txt
I tried to make the build of Android so simple that simple enough:
1) **git clone https://github.com/minetest/minetest**
2) **cd minetest/build/android**
3) **./gradlew assemblerelease**
(./gradlew.bat assemblerelease for Windows).
Or you can use Android Studuio (connect the phone to the PC, enable USB debugging and click Build & Run).

This will prepare the environment, clone git with precompiled libraries, the assembly of which is impossible without Cygwin, a large number of dependent packages and a dozen hours of configuration testing.

In order for the assembly instructions for the content in just 3 simple steps, to be universal for all platforms and the assembly was very fast even on a weak PC (try amalg luajit assembly on a PC with 2GB of RAM!), This turned out to be necessary.

Reply:
To build MT, the following libraries are required (armeabi-v7a and arm64-v8a):
1) Curl
2) Freetype2
3) Irrlicht (with some android specific patches!)
4) LuaJIT-2.1
5) OpenAL-Soft 1.19.1 (1.20.* has problems with Android)
6) libogg
7) libvorbis (with some android specific patches!)
8) libiconv (pre-configuration on linux required, no pre-assembly required)
9) sqlite3 (no pre-assembly required)

@SmalllJoker, do you want to be the first to build this shit on Windows and macOS for Android?

Reply:
https://github.com/minetest/minetest_android_deps_binaries

Please adjust the code accordingly.

Reply:
Done